client-src/elements/chromedash-gantt.js

```diff
@@ -128,6 +133,26 @@ class ChromedashGantt extends LitElement {
     return cellsOnRow;
   }
 
+  // Get lists of all dev trial, origin trial, and shipping stages
+  // associated with the feature.
+  getByStageType() {
+    const dtStages = [];
+    const otStages = [];
+    const shipStages = [];
+
+    for (const stage of this.feature.stages) {
+      if (STAGE_TYPES_DEV_TRIAL.has(stage.stage_type)) {
+        dtStages.push(stage);
+      } else if (STAGE_TYPES_ORIGIN_TRIAL.has(stage.stage_type)) {
+        otStages.push(stage);
+      } else if (STAGE_TYPES_SHIPPING.has(stage.stage_type)) {
+        shipStages.push(stage);
+      }
+    }
+
+    // TODO(DanielRyanSmith): Fix this to support multiple stages.
+    return [dtStages[0], otStages[0], shipStages[0]];
```

Comment:
Could one of these lists be empty?  I suppose that returning `undefined` is OK.

Reply:
Yes, my first thought process was to check if any list was empty, but JS will just return undefined for out of bounds accesses. If this code was supposed to be permanent, I'd make this more explicit, but ideally we update this to handle the new schema accurately anyway.